system/lib/wasmfs/syscalls.cpp

```diff
@@ -540,6 +545,9 @@ static long doMkdir(char* path, long mode, backend_t backend = NullBackend) {
   }
 
   auto base = pathParts.back();
+  if (base.size() >= WASMFS_NAME_MAX) {
```

Comment:
It's strange that this comparison is `>=`. The name `WASMFS_NAME_MAX` makes me think that names of that length would be allowed (but longer names would not be allowed).

Reply:
I could change it to `255`. On Linux, it seems making a file with `256` characters is not permitted.